samples/sensor/bme280/src/main.c

```diff
@@ -9,25 +10,37 @@
 #include <devicetree.h>
 #include <drivers/sensor.h>
 
-#define BME280 DT_INST(0, bosch_bme280)
+/*
```

Comment:
This seems fine for now; if we're going to convert all samples maybe this could go into a sample support library, with a declaration like:

```
const struct device *sample_device_validate(const struct device *dev);
...
const struct device *dev = sample_device_validate(DEVICE_DT_GET_ANY(bosch_bme280));
```
so we don't have to repeat 20 lines of boilerplate in each sample.

Alternatively, the validation is something real applications need to do (though probably without quite so much verbose output).  Something to discuss later.

Reply:
> Alternatively, the validation is something real applications need to do (though probably without quite so much verbose output). Something to discuss later.

Something like this, perhaps: https://gist.github.com/mbolivar-nordic/f1a771d68e2f70cdead17a944c049349

Reply:
Would actually be nice to have this as well in this PR so it could directly serve as a basis for other samples conversion and avoid multiple rework steps. This would be, from now on, _the_ way to access devices, period.
But won't block for this as, this is already a great improvment.

Reply:
> Something like this, perhaps: https://gist.github.com/mbolivar-nordic/f1a771d68e2f70cdead17a944c049349

Not really what I meant by "Alternatively, ...".

The sample has to do verbose validation because it's being run by people who aren't sure whether things are set up right, and they need clear indication when something goes wrong.  The argument against putting the boilerplate in every sample is that those checks are really verbose, distract from from the purpose of the sample, and make it look like Zephyr is more complicated than it is.

In real code application I personally don't need a helper.  I particularly don't want one that makes policy on how failures are handled.  I'd be checking the conditionals myself and deciding whether it's just turning off a feature, disabling the driver, or causing a fatal error (most of which I wouldn't have conditional on something like `CONFIG_ASSERT`).

>  avoid multiple rework steps

I really hope we can coordinate the conversion and provide any shared sample-specific API while we discuss #32139 *before* anybody dives in and starts doing mass conversion.

Reply:
Thanks @erwango .  Yes, I'd like to defer this change for now to get the existing improvements in when the merge window opens.

It will be easy to fix up this sample once we've got more samples using DEVICE_DT_GET_ANY and we have agreement on how to avoid boilerplate, which IMO is best done in a separate PR.

Reply:
> In real code application I personally don't need a helper. I particularly don't want one that makes policy on how failures are handled.

For sure, I think we all feel this way. I've reworded #32139 to try to make it clearer that I'm talking about the "shut up and give me a device" use case that users who are using samples to evaluate drivers run into.